code/go/0chain.net/smartcontract/dbs/event/snapshot.go

```diff
@@ -0,0 +1,18 @@
+package event
+
+// swagger:model Block
+type Snapshot struct {
+	Round           int64  `gorm:"primaryKey;autoIncrement:false" json:"round"`
+	BlockHash       string `json:"block_hash"`
+	MintTotalAmount int64  `json:"mint_total_amount"`
+}
+
+func (edb *EventDb) GetRoundsMintTotal(from, to int64) (int64, error) {
+	var total int64
+	return total, edb.Store.Get().Model(&Snapshot{}).Where("round between ? and ?", from, to).Select("sum(mint_total_amount)").Scan(&total).Error
+}
+
+func (edb *EventDb) addOrUpdateTotalMint(mint *Mint) error {
```

Comment:
use upsert here https://gorm.io/docs/create.html#Upsert-x2F-On-Conflict

```go
// Update columns to new value on `id` conflict
db.Clauses(clause.OnConflict{
  Columns:   []clause.Column{{Name: "round"}},
  DoUpdates: clause.AssignmentColumns([]string{"mint_total_amount"}),
}).Create(&mint)
```

Reply:
With gorm 1.9.x or below, it's more effecient to update first, then insert when not existed.
But both doing the same thing.